### PR TITLE
Add background layer selector as 'drop-down' button in desktop apps

### PR DESCRIPTION
**Pull request by adube**
This PR is the replacement of #1291.  It adds the `gmf-backgroundlayerselector` inside both desktop applications using contextual menu customized to look like a button.

Original commit: https://github.com/pgiraud/ngeo/commit/9f6b0a10e427be6ba36ee625e935aedf62e050ec

## Todo

 * [x] review

## Live demos

 * https://adube.github.io/ngeo/bls-in-desktop-apps/examples/contribs/gmf/apps/desktop/
 * https://adube.github.io/ngeo/bls-in-desktop-apps/examples/contribs/gmf/apps/desktop_alt/

### CODE REVIEW

**Comment by adube** [comment]
Ready for review.

File: contribs/gmf/apps/desktop/index.html [hunk 1]
```diff
@@ -99,6 +99,17 @@
           gmf-search-coordinatesprojections="mainCtrl.searchCoordinatesProjections"
           gmf-search-clearbutton="true">
         </gmf-search>
+        <div class="gmf-backgroundlayerbutton btn-group dropup">
+          <button
+              class="btn btn-default"
+              data-toggle="dropdown">
+            <img src="https://cloud.githubusercontent.com/assets/319774/13696817/1cc9a7ee-e769-11e5-85c8-a1ea5ff21a5d.png" alt="" />
```
Comment (sbrunner) [inline contribs/gmf/apps/desktop/index.html]: We don't want to add the image in the repository?

File: contribs/gmf/less/desktop.less [hunk 2]
```diff
@@ -415,3 +415,28 @@ gmf-featurestyle {
...
   width: 20rem;
 }
+
+
+/**
+ * Background layer button (selector)
+ */
+.gmf-backgroundlayerbutton {
+  bottom: 4rem;
+  left: 1em;
```
Comment (sbrunner) [inline contribs/gmf/less/desktop.less]: We should use rem for all units ...

File: contribs/gmf/apps/desktop/index.html (hunk 1 above)
Comment (adube) [inline contribs/gmf/apps/desktop/index.html]: Please, see with @pgiraud or @fredj .  I don't know the answer to that question.

File: contribs/gmf/less/desktop.less (hunk 2 above)
Comment (adube) [inline contribs/gmf/less/desktop.less]: Oops, typo.  Good catch, I'm on it right away.

File: contribs/gmf/apps/desktop/index.html (hunk 1 above)
Comment (pgiraud) [inline contribs/gmf/apps/desktop/index.html]: We should definitely use a local image. It was OK to use a URL only in a prototype context.

Comment (adube) [inline contribs/gmf/apps/desktop/index.html]: Understood.  I'm on it.

File: contribs/gmf/less/desktop.less (hunk 2 above)
Comment (fredj) [inline contribs/gmf/less/desktop.less]: Add the less variable must be used (`@app-margin`, ...)

**Comment by adube** [comment]
Image isn't shown on github.io.

**Comment by pgiraud** [comment]
> Image isn't shown on github.io.

It doesn't show in built mode I think. You have to use the same strategy as for the geomapfish logo.
https://github.com/camptocamp/ngeo/blob/master/Makefile#L465-L468

**Comment by sbrunner** [comment]
@adube you should do some thing like that https://github.com/camptocamp/ngeo/pull/1291#issuecomment-221872271, and what about put the image in `contribs/gmf/images`?

**Comment by adube** [comment]
O~kay!  All raised issues have been fixed.  Let me know if you need me to fix anything else.

**Comment by adube** [comment]
@sbrunner Yeah, that's what has just been done.  The image was put in the `image` directories of the applications.

**Comment by sbrunner** [comment]
OK :-)

**Comment by pgiraud** [comment]
This looks good to me. Can you please squash the commits?
The last thing that I dislike are the margins in the selector itself but it should be addressed in an other issue. I'll open one.

**Comment by adube** [comment]
"Squashing squashing..."

**Comment by adube** [comment]
Quick unrelated question: I was told a few weeks ago that it wasn't required anymore to squash commits into once since it's now possible to automatically do so when merging on Github.  I used to always squash the commits, but was told that I wasn't forced to do so anymore.  Is that still the case ?  I'm okay with doing one or the other, but let's agree on just one way to go.

I think I'd slightly prefer "not squashing" them manually.  This makes it easier for reviewers.  If one asks: "please, fix this" then he or she can immediately see in the next commit that it has been fixed without having to look at the whole code again.

What do you think ?

**Comment by adube** [comment]
Is this ready to be merged ?

**Comment by fredj** [comment]
ready, I will merge when travis is green

**Pull request merged**